### DISCUSSION TITLE
Welcome to Asrock-B460m-ITX-AC-OC-EFI Discussions!

### DISCUSSION THREAD

**Discussion by ansonliao** [ansonliao/Asrock-B460m-ITX-AC-OC-EFI#15] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by bigcolor-eddie** [comment]
FYI: The new BIOS 1.6 has "CFG Lock" on/off function now.

**Reply by ansonliao** [reply]
Asrock B460M-ITX-AC's BIOS should has the `CFG Lock` when the first release, should be not the new feature after BIOS 1.6.

**Reply by bigcolor-eddie** [reply]
why I said that because I saw the code in the config file with the following codes:

		<key>Quirks</key>
		<dict>
			<key>AppleCpuPmCfgLock</key>
			<true/>
			<key>AppleXcpmCfgLock</key>
			<true/>

so it makes me confused

**Reply by ansonliao** [reply]
You can set this configuration as `false` if you disabled `CFG Lock` in your BIOS.
In OC, there is no problem enabling `AppleCpuPmCfgLock` and disabled BIOS `CFG Lock` together.
You can only select one option to work, OC or BIOS is fine.

**Comment by wwbingo** [comment]
how do you use the BCM94360CS2 on the b460m-itx/ac, it seems that no interface for the wireless card

**Reply by ansonliao** [reply]
Please check the images of the blog below.
https://icyleaf.com/2019/03/asrock-z390-gaming-itx-install-hackintosh-tutorial/

**Reply by wwbingo** [reply]
we can use BCM94352Z instead of BCM94360CS2

**Reply by ansonliao** [reply]
You can use any WiFi card when you can make the card work, not only these two cards

**Comment by wwbingo** [comment]
Thank you for your EFI and the hardware specification, I had power the macOS perfect.

**Comment by wolf4914** [comment]
Hi! Are you planning to upgrade  to 12.5.1 in the near future by any chance? I am about to install it but not sure if it is going to work

**Reply by ansonliao** [reply]
My computer was tidied up, I am afraid can't update the EFI recently.

**Comment by loupe123** [comment]
请问还会继续维护吗

**Comment by ansonliao** [comment]
@wolf4914 @loupe123 ,
1. Upgraded OC version to 0.8.7
2. Supported macOS Ventura 13.1

Release with notes:
- https://github.com/ansonliao/Asrock-B460m-ITX-AC-OC-EFI/releases/tag/3.0

**Comment by peminator** [comment]
im trying like 20 times, for ASRock B460M-ITX,  and still no boot to installer, should i use exactly oc 0.8.7 or can be used also more recent version ? what is the steps? 
1) oc prepare basic usb, 
2) copy over ALL your files, 
3) plist clean snapshot 
4) smbios  generate serial
and thats all? Have exactly same cpu  (i5 10500, no gpu added, 16GB ram). Should i also overwrite BOOT/BOOTX64.efi etc ? 
having a final zip with all prepared to just download the apple files for ventura would be so helpful, i keep trying and always installer somehow fails (black screen before showing boot menu so i did something wrong, but what) 

iGPU Patching info - that to be done after install ? what app? im missing a full step-by-step, maybe some video would help

mentioned in notice device-id  - what where and how should i do it ?

**Comment by peminator** [comment]
hi, i have no added gpu, only the integrated in cpu (UHD Graphics 630)

**Reply by ansonliao** [reply]
Hi @peminator , what is your BIOS configuration?

**Reply by peminator** [reply]
well mostly i tried after reset to defaults, not sure if need secure boot enabled or disabled, or any special setting. 
How do i tell you all the settings? Post screens as link to some album? There are many sub-pages imho, not sure if post all, or what pages matter. Also, i think i may have older firmware than the last current version. I can try update, if it can be done from windows (i know it has also live update over internet, but requires LAN, im connected only using wifi)

**Reply by peminator** [reply]
also what is the correct way to use this repo? only use the files here by default, just update the serial using the [GenSMBIOS](https://github.com/corpnewt/GenSMBIOS) and download the  chosen version to com.apple.recovery.boot ? anything else ?  The point in notices about device-id, can be done after install ? Maybe make a short step-by-step for noobs. Updated Uefi firmware today to 1.60, still no boot just black screen.... maybe also replace release oc to debug, so see if no boot, what reason?

**Reply by ansonliao** [reply]
Hi @peminator , 
1. For the BIOS settings, please check: https://github.com/ansonliao/Asrock-B460m-ITX-AC-OC-EFI#bios-settings
2. How to use this repo, after you created your boot USB of macOS installation:
  2.1 Copy the EFI folder to the root directory of your EFI partition of the boot USB drive.
  2.2 Generate the SMBIOS to get the valid `System Serial Number`, `System UUID`, and `MLB`, and fill them to `config.plist` -> `PlatformInfo` -> `DataHub - Generic - PlatformNVRM`, as the screenshot below.
3. Then boot your machine from the EFI partition of your boot USB of the macOS installation.

<img width="1337" alt="image" src="https://github.com/ansonliao/Asrock-B460m-ITX-AC-OC-EFI/assets/6060224/c13842b0-2f81-4fd1-ba90-ca7e6f4bf250">

**Reply by peminator** [reply]
so basically feel like idiot for asking, but all i manage to get i boot to empty black screen nad nothing helps. Maybe what do u mean by the step "after you created your boot USB of macOS installation" ?? 

preparing USB means just:
1) create empty Fat32 usb 
2) use the downloader and place downloaded "com.apple.recovery.boot" to USB root, 
3) copy your EFI folder to root of USB
4) use GenSMBIOS to add UUID etc to plist

Thats all? I keep getting to black screen, nothing more, getting crazy 
I dont know what should be  MLB, but GenSMBIOS seems has filled that value too

btw what is the editor u have screenshoted before? I use ProperTree under windows, from OC 0.8.7

![image](https://github.com/ansonliao/Asrock-B460m-ITX-AC-OC-EFI/assets/1151036/5d293646-8a94-4c03-a9f1-ab206068a2ea)

![image](https://github.com/ansonliao/Asrock-B460m-ITX-AC-OC-EFI/assets/1151036/5091f054-3f54-44a2-baaf-b9685c6dd9ea)

**Reply by ansonliao** [reply]
Hi @peminator ,
Do you read and follow the USB installation creation of OpenCore official document?
https://dortania.github.io/OpenCore-Install-Guide/installer-guide/mac-install.html#setting-up-the-installer

**Comment by peminator** [comment]
I was not sure which way should I use. So by this manual, I start on the link u send, and can stop before "Adding The Base OpenCore Files" bc i add yours instead. 

Correct? 

Will try on weekend and report back here on result. Does it matter anyhow which version of OS i choose? Id love to try Ventura,
dont remember which version is there available (if i remember from last time, there only is 13.4, no 13.1 or so in the downloader

**Reply by ansonliao** [reply]
Hi @peminator ,
Are you creating your installer USB disk in Windows or Mac?
If you creating the installer USB disk in Windows, maybe you can refer to some tutorials like below:
- https://ichumz.com/create-macos-ventura-bootable-usb-on-windows/

**Reply by peminator** [reply]
Thanks will try. At the moment have both available - macbook air (2018) and a windows pc (on that asrock mb).

**Reply by ansonliao** [reply]
If you have macOS machine, I think it's easy to create the USB bootable USB drive.

**Comment by ants83** [comment]
Hi @ansonliao, is there a specific reason why you chose the `Macmini8,1` SMBIOS and not the `iMac20,1` one, as specified in [Dortania Guide](https://dortania.github.io/OpenCore-Install-Guide/config.plist/comet-lake.html#platforminfo)?

Thanks.

**Reply by ansonliao** [reply]
Hi @ants83 , my specification is no dGPU, only iGPU of Intel CPU built.

**Reply by ants83** [reply]
In Opencore Install Guide I didn't find any reference to the `iMac20,1` SMBIOS being recommended specifically for dGPU configurations. I believe the choice is related uniquely to the `iMac20,1` ([iMac 27" 2020](https://everymac.com/systems/apple/imac/specs/imac-core-i5-3.1-6-core-27-inch-retina-5k-2020-specs.html)) being equipped with a 10th Gen Comet Lake processor.

At present I am running a iGPU-only setup and I am using your exact EFI configuration (by the way, thanks!) but with the `iMac20,1` SMBIOS and everything seems to be ok. It would be interesting to understand the differences.

**Reply by ansonliao** [reply]
Please check the document: https://dortania.github.io/OpenCore-Install-Guide/extras/smbios-support.html#choosing-the-right-smbios
I think if running the high requirement to dGPU job, maybe can see the different when chosen `iMac` smbios and no dGPU installed.
